### PR TITLE
#163426613 Add ESlint configuration

### PR DESCRIPTION
**Pull request by mrkingz**
#### What does this PR do?
Add ESlint configuration to enable linting
#### Description of Task to be completed?
Enables linting to find problematic patterns or codes that don’t adhere to certain style guidelines
#### What are the relevant pivotal tracker stories?
[#163426613](https://www.pivotaltracker.com/n/projects/2240406/stories/163426613)

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Dependency directory
+node_modules
```
Comment (benfluleck) [inline .eslintignore]: No new line here

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        }],
+        "require-jsdoc": ["error", {
+            "require": {
+                "FunctionDeclaration": true,
+                "MethodDefinition": true,
+                "ClassDeclaration": true
+            }
+        }]       
+    }
+}
```
Comment (benfluleck) [inline .eslintrc.json]: New line needed here

**Comment by benfluleck** [comment]
There are too many commit messages in this PR. Please squash them @mrkingz

**Review by eokwukwe — CHANGES_REQUESTED** [review]
Edit PR message. The relevant PT story part needs to only contain the story ID and it should be made to link to the PT story.

File: .eslintignore (hunk 1 above)
Comment (mrkingz) [reply .eslintignore]: Fixed. Thanks

File: package.json [hunk 3]
```diff
@@ -3,8 +3,28 @@
...
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
+  "engines": {
+    "node": ">=6.0.0"
+  },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "start": "node server/index.js",
+    "test": "exit 0",
+    "lint": "./node_modules/.bin/eslint 'server/**/*js'",
```
Comment (benfluleck) [inline package.json]: install eslint globally,  I expect `eslint '**/*js`
Why is this being done one by one. it should just be one command to fix
`    "lint:fix": "prettier-eslint '**/*js --write",`
Why is there so much consideration for json files, where are they coming from?

**Review by benfluleck — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (mrkingz) [reply package.json]: Fixed

**Comment by benfluleck** [comment]
@mrkingz  please squash and amend your  commits, there are too many

File: .gitignore [hunk 4]
```diff
@@ -28,13 +28,19 @@ build/Release
...
 node_modules
 
+
 # Optional npm cache directory
 .npm
 
 # Optional REPL history
 .node_repl_history
 
+<<<<<<< HEAD
```
Comment (benfluleck) [inline .gitignore]: This happens when you have ignored your conflicts please fix

File: .gitignore [hunk 5]
```diff
@@ -28,13 +28,19 @@ build/Release
...
 # Coverage Output
 .nyc_output
 .coveralls.yml
- 
\ No newline at end of file
+ 
+=======
+# coveralls
+.coveralls.yml
+>>>>>>> chore (Setup): configure ESlint - Add ESlint configuration [Finishes #163426613]
```
Comment (benfluleck) [inline .gitignore]: This should not be showing up in your branch

**Review by benfluleck — CHANGES_REQUESTED** [review]

**Pull request closed without merging**